docs/manual/upgrades.md

```diff
@@ -0,0 +1,129 @@
+# FoundationDB Upgrades
+
+## Overview
+
+The operator supports to upgrade FDB clusters by adjusting the `version` setting in the `FoundationDBCluster` spec.
+It also offers limited support for downgrades, the limitation is that the operator only supports downgrades to an older patch version of the running FDB version.
+Upgrades of FoundationDB running on Kubernetes are challenging and the operator provides a way to handle a lot of the complexity on itself.
+The document will describe the upgrade process in details and some cases the operator cannot handle currently.
+
+FoundationDB upgrades for major and minor versions are incompatible and require that all processes are updated in a short time period.
+
+### Upgrade Process Details
+
+This document describes the upgrade process based on the operator version `v1.14.0` for older versions there might be a slight difference in how the operator is handling the upgrades.
+In addition to that we are currently working on reducing the complexity for patch upgrades to reduce the upgrade steps to the [Recreation of old Pods](#recreation-of-pods-phase) phase.
+
+#### Pre Upgrade Check Phase
+
+Before the upgrade is initiated the operator will perform a compatibility check for all clients of this cluster.
+This check will ensure that a cluster is only upgraded if all clients support the new version, for patch upgrades this check is skipped.
+More information about this reconciler can be found in the [technical design](technical_design.md#checkclientcompatibility).
+Clients not supporting the new version will be reported in the logs of the operator with the message `Deferring reconciliation due to unsupported clients` and in addition the operator will emit a Kubernetes event.
+This prevents upgrading a database until all clients have been updated with a compatible client library.
+
+#### Staging Phase
+
+This phase ensures that all Pods are in a state that the operator can restart all `fdbserver` processes and they will be restarting with the new FDB version.
+The first step is the `UpdateSidecarVersions`, this subreconciler updates the image of the `foundationdb-kubernetes-sidecar` container to match the desired FDB version.
+Updating the image of a single container doesn't require the recreation of the Pod and keeps the main container that runs the `fdbserver` process running.
+The sidecar container will copy the new binary inside the main container over a shared volume.
+In the main container the new binaries will be present in `/var/dynamic-conf/bin/$fdb_version`, this version will be used once the `fdbserver` process is restarted.
+The `UpdatePodConfig` subreconciler will ensure that all Pods have the new configuration present, this configuration ensures that the `fdbmonitor` will restart processes with the new binary present in the shared volume.
+Once all Pods have the new configuration and binaries present the operator can move to the next phase.
+
```

Comment:
Same question here: we don't advance to the Restart phase if not all pods have the new configuration/binaries present, correct?

Reply:
Correct.